Code review (#1)
StuartCowley/music-library-1: README.md

# Music Library 🎶

**First project with [Manchester Codes](https://www.manchestercodes.com/) BACKEND module**

The purpose of this project was to apply all learned aspects together in one project by building an API (Application Programming Interface) app to manage a database, a music library.

The project followed TDD principles by writing a failing test and write the code to pass the test. This time we used Mocha supertest, an integration test which checks that app is communicating with the database.

---

> **Consepts**

Author: ```suggestion
**Concepts**
```

Reviewer: Lots of spelling mistakes in the Readme file. Teaches me to take time to proofread!

- Database Design
- SQL
- MySQL
- CRUD (CREATE, READ, UPDATE, DELETE) Operations

---

> **Programming langages and technologies used**

Author: ```suggestion
**Programming languages and technologies used**
```

- Express, NodeJS
- MySQL
- Mocha /Chai supertest
- Docker
- Postman
- MySQL workbench

---

**Set up**

> To be able to run this application follwing environment needs to be installed:

Author: ```suggestion
To be able to run this application following environment needs to be installed:
```

[Docker](https://docs.docker.com/get-docker/)

[MySQLWorkbench](https://www.mysql.com/products/workbench/)

[Postman](https://www.postman.com/)

```
npm init -y
```

```
npx eslint --init
```

```
npm i -S express
```

```
npm i -D nodemon
```

```
npm i -D dotenv
```

```
npm i -S mysql2
```

```
npm i -D mocha chai supertest

> ```

Author: Given that all these packages are already included in the `package.json` (and will be installed to the version specified in there when running `npm i`) these commands do not need to be run or included here. It would be worth including a line simply stating that running `npm i` is enough to get the project up and running from a code perspective. 
Also some notes on needing a database to be setup would be a good idea

---

**Project status**

Completed and deployed with Heroku [here](https://music-library-helid.herokuapp.com/artist). Use Postman as a client to insert an artist to the database.

Possible ways to continue this project:

- extend the app to include album table.

- connect artist id to album

> I have left some commented out code which is not working just as a notes for myself.

Author: This is not a good idea in general, there is a more detailed explanation on the code in question

---

**Comments**

> The project has streghten my knowledge of the complex _BACKEND_ development, _ODD (Object oriented programming)_ by keeping ROUTES, CONTROLLERS and SERVICES separate and of course again highlighed the importance of the testing environment.

Author: Why the block capitals and inconsistent italics? At first glance not sure what the significance of this formatting is

```suggestion
The project has strengthen my knowledge of the complex _BACKEND_ development, _OOP (Object Oriented Programming)_ by keeping ROUTES, CONTROLLERS and SERVICES separate and of course again highlighted the importance of the testing environment.
```
